src/charm.py

```diff
@@ -180,6 +221,8 @@ def _cli_args(self):
 
         return args
 
+    # TODO This should throw and be handled. RIght now, we just
```

Comment:
How would throwing an exception be handled ? Only reasonable options I can think of is either set the charm in blocked state or log an error and let the charm continue running since at least it will still be scraping metrics from all other relations.